File: Army(c++)/Warriors/Berserker.cpp
#include "Berserker.hpp"

Berserker::Berserker(const std::string& name, int hp, int dmg) 
    : Unit(name, hp, dmg, new SoldierState(this))
    , warrior(WarriorID::BERSERKER) {}

Berserker::~Berserker() {}

WarriorID Berserker::getWarriorID() const {
    return this->warrior;
}

void Berserker::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

File: Army(java)/Warriors/Berserker.java
package Warriors;

import EnumID.*;
import Basic.*;
import State.*;

public class Berserker extends Unit {
    private WarriorID warrior;
    
    public Berserker(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.BERSERKER;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }
}
